### DISCUSSION TITLE
In clipboard copy paste, the custom function of the object can't be copied, how do I do it?

### DISCUSSION THREAD

**Discussion by james-ep01** [clientIO/joint#2808] [Q&A]
### Introduction

Below is the class I made and am using. **I'm using my own variables and functions.**

![image](https://github.com/user-attachments/assets/c7e30a80-20c7-4349-bae9-d4a2c45995d2)

The code below is a copy and paste code.
![image](https://github.com/user-attachments/assets/78a51e92-1c0e-48b4-a74b-e76781fdd543)

I printed it out as console in copy and paste, but you can see that the custom variables are pasted as they are, **but the custom 
 function disappears when pasted.**

What is the problem?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
You can not have the clipboard's `useLocalStorage` option enabled. The clipboard is serialized and save to the local storage. All functions are stripped.

It's the same as you would serialize the graph and send to backend with `JSON.stringify(graph.toJSON())`.

Why don't you add the method to the prototype of the class?

```ts
const Shape = dia.Element.extends({
    setPower: function() { /* ... */ } 
});

const shape = new Shape();
shape.setPower();
```

**Reply by james-ep01** [reply]
![image](https://github.com/user-attachments/assets/1507691a-7e90-45b3-ba82-8e75f2b1d23a)

Oh, you use localStorage by json serializing the clipboard copy function..
I can't copy my function, so I'll have to use another method.
This is because copied elements cannot use functional functions.

The objects that were exported to json were also json serialized, so there was no my function...

But why can't ToolVIew button or Highlight be copied and exported together?

**Reply by kumilingus** [reply]
Because neither tool views nor highlighters are saved as model attributes (at least by default).

However, you can introduce a custom model attribute, e.g. `highlighted`. When an element with `highlighted=true` is added to the graph, you can add a HighlighterView to its view.
